ledger/participant-integration-api/src/main/scala/platform/indexer/StandaloneIndexerServer.scala

```diff
@@ -27,73 +27,73 @@ final class StandaloneIndexerServer(
   private val logger = ContextualizedLogger.get(this.getClass)
 
   override def acquire()(implicit context: ResourceContext): Resource[ReportsHealth] = {
+    val flywayMigrations =
+      new FlywayMigrations(
+        config.jdbcUrl,
+        config.enableAppendOnlySchema,
+        additionalMigrationPaths,
+      )
     val indexerFactory = new JdbcIndexer.Factory(
       ServerRole.Indexer,
       config,
       readService,
       servicesExecutionContext,
       metrics,
       lfValueTranslationCache,
-      additionalMigrationPaths,
     )
     val indexer = RecoveringIndexer(
       materializer.system.scheduler,
       materializer.executionContext,
       config.restartDelay,
     )
+
+    def startIndexer(
+        migration: Future[Unit],
+        initializedDebugLogMessage: String = "Waiting for the indexer to initialize the database.",
+    ): Resource[ReportsHealth] =
+      Resource
+        .fromFuture(migration)
+        .flatMap(_ => indexerFactory.initialized.acquire())
+        .flatMap(indexer.start)
+        .map { case (healthReporter, _) =>
+          logger.debug(initializedDebugLogMessage)
+          healthReporter
+        }
+
     config.startupMode match {
       case IndexerStartupMode.MigrateAndStart =>
-        Resource
-          .fromFuture(
-            indexerFactory
-              .migrateSchema(config.allowExistingSchema)
-          )
-          .flatMap(startIndexer(indexer, _))
-          .map { healthReporter =>
-            logger.debug("Waiting for the indexer to initialize the database.")
-            healthReporter
-          }
+        startIndexer(
+          migration = flywayMigrations.migrate(config.allowExistingSchema)
+        )
+
       case IndexerStartupMode.ResetAndStart =>
-        Resource
-          .fromFuture(indexerFactory.resetSchema())
-          .flatMap(startIndexer(indexer, _))
-          .map { healthReporter =>
-            logger.debug("Waiting for the indexer to initialize the database.")
-            healthReporter
-          }
+        startIndexer(
+          migration = flywayMigrations
+            .reset()
```

Comment:
This startup mode is used from sandbox (sandboxnext).
As discussed with @gerolf-da , it is acceptable now to switch to complete wiping, since sandbox is not participating in performance intensive testing anymore.
The motivation to switch to flywaMigrations.reset() here is to decouple JdbcIndexer from reset responsibilities.

Are you aware of some use-case/usage, which would motivate the usage of the reset service still?
cc @meiersi-da @rautenrieth-da @mziolekda @miklos-da @stefanobaghino-da @oliverse-da @SamirTalwar-DA